src/serialisation.rs

```diff
@@ -0,0 +1,235 @@
+use dusk_bls12_381::{G1Affine, G2Affine, Scalar};
+use thiserror::Error;
+
+/// Defines all of the possible Serialisation errors
+#[derive(Error, Debug)]
+pub enum SerialisationErrors {
+    #[error("There are not enough bytes to perform deserialisation")]
+    NotEnoughBytes,
+    #[error("Cannot decompress point, as it is not in a canonical format")]
+    PointMalformed,
+    #[error("Cannot deserialise scalar, as it is not in a canonical format")]
+    ScalarMalformed,
+}
+
+/// Reads n bytes from slice and returns the n bytes along with the rest of the slice
+pub fn read_n(n: usize, bytes: &[u8]) -> Result<(&[u8], &[u8]), Error> {
+    if bytes.len() < n {
+        return Err(SerialisationErrors::NotEnoughBytes.into());
+    }
+    let bytes32 = &bytes[0..n];
+    let rest = &bytes[n..];
+    Ok((bytes32, rest))
+}
+
+/// Reads 32 bytes and converts it to a Scalar
+/// Returns the remaining bytes
+pub fn read_scalar(bytes: &[u8]) -> Result<(Scalar, &[u8]), Error> {
+    let (bytes32, rest) = read_n(32, bytes)?;
+    let mut arr32 = [0u8; 32];
+    arr32.copy_from_slice(bytes32);
+    let scalar = Scalar::from_bytes(&arr32);
+    if scalar.is_none().into() {
+        return Err(SerialisationErrors::ScalarMalformed.into());
+    }
+    Ok((scalar.unwrap(), rest))
+}
+/// Writes a Scalar into a mutable slice
+pub fn write_scalar(scalar: &Scalar, bytes: &mut Vec<u8>) {
+    bytes.extend_from_slice(&scalar.to_bytes());
+}
+
+/// Reads 48 bytes and converts it to a G1Affine
+/// Returns the remaining bytes
+pub fn read_g1_affine(bytes: &[u8]) -> Result<(G1Affine, &[u8]), Error> {
+    let (bytes48, rest) = read_n(48, bytes)?;
+    let mut arr48 = [0u8; 48];
+    arr48.copy_from_slice(bytes48);
+    let g1 = G1Affine::from_compressed(&arr48);
+    if g1.is_none().into() {
+        return Err(SerialisationErrors::PointMalformed.into());
+    }
+    Ok((g1.unwrap(), rest))
+}
+/// Reads 48 bytes and converts it to a Commitment
+/// Returns the remaining bytes
+pub fn read_commitment(bytes: &[u8]) -> Result<(Commitment, &[u8]), Error> {
+    let (g1, rest) = read_g1_affine(bytes)?;
+    Ok((Commitment::from_affine(g1), rest))
+}
+/// Writes a G1Affine into a mutable slice
+pub fn write_g1_affine(affine: &G1Affine, bytes: &mut Vec<u8>) {
+    let bytes48 = affine.to_compressed();
+    bytes.extend_from_slice(&bytes48);
+}
+/// Writes a Commitment into a mutable slice
+pub fn write_commitment(commitment: &Commitment, bytes: &mut Vec<u8>) {
+    write_g1_affine(&commitment.0, bytes)
+}
+
+/// Reads 96 bytes and converts it to a G2Affine
+/// Returns the remaining bytes
+pub fn read_g2_affine(bytes: &[u8]) -> Result<(G2Affine, &[u8]), Error> {
+    let (bytes96, rest) = read_n(96, bytes)?;
+    let mut arr96 = [0u8; 96];
+    arr96.copy_from_slice(bytes96);
+    let g2 = G2Affine::from_compressed(&arr96);
+    if g2.is_none().into() {
+        return Err(SerialisationErrors::PointMalformed.into());
+    }
+    Ok((g2.unwrap(), rest))
+}
```

Comment:
I belive serde is implemented for all of the `bls12_381` structures. It's fine anyway if you want to use your own method.